packages/base/lua/Profiler.lua

```diff
@@ -0,0 +1,282 @@
+		if block and block.running then
+			customWarning(string.format("Block '%s' has already been started.", block.name))
+		elseif not block then
+			block = createBlock(name)
+			self.blocks[name] = block
+		end
+
+		if self.currentName and self.currentName ~= name then
+			self:stop(self.currentName)
+		end
+
+		self.currentName = name
+		block.count = block.count + 1
+		block.running = true
+	end,
+	--- Return the current block.
+	-- @usage Profiler():start("block")
+	-- print(Profiler():current().name) -- block
+	-- Profiler():stop("block")
+	current = function(self)
+		return self.blocks[self.currentName]
+	end,
+	--- Return how many times a given block has started.
+	-- @arg name A string with the block name. If the name is not informed, then it returns the count of the current block.
+	-- @usage Profiler():start("block")
+	-- print(Profiler():count("block")) -- 1
+	-- Profiler():stop("block")
+	count = function(self, name)
+		optionalArgument("name", "string", name)
+		local block = self.blocks[name or self.currentName]
+		if not block then
+			customError(string.format("Block '%s' not found.", name or self.currentName))
+		end
+
+		return block.count
+	end,
+	--- Return how much time was spent on the block.
+	-- It returns two representations: a string with a human-like representation of the time
+	-- and a number with the time in seconds.
+	-- @arg name A string with the block name. If the name is not informed, then it returns the uptime of the current block.
+	-- @usage Profiler():start("block")
+	-- Profiler():stop("block")
+	-- stringTime, numberTime = Profiler():uptime("block")
+	uptime = function(self, name)
+		optionalArgument("name", "string", name)
+		local block = self.blocks[name or self.currentName]
+		if not block then
+			customError(string.format("Block '%s' not found.", name or self.currentName))
+		end
+
+		local time = block:uptime()
+		return timeToString(time), time
+	end,
+	--- Stop to measure the time of a given block. It also returns how much time was spent with the block
+	-- in two representations: a string with a human-like representation of the time and a number with the time in seconds.
+	-- @arg name A string with the block name. If the name is not informed, then it stops and return the uptime of the current block.
+	-- @usage Profiler():start("block")
+	-- stringTime, numberTime = Profiler():stop("block")
+	stop = function(self, name)
+		optionalArgument("name", "string", name)
+		if (name == "main" or not name and self.currentName == "main") and getn(self.blocks) == 1 then
+			customWarning("The block 'main' cannot be stopped.")
+			return self:uptime(name)
+		end
+
+		local block = self.blocks[name or self.currentName]
+		if block and block.running then
+			block.running = false
+			block.endTime = sessionInfo().time
+		elseif not block then
+			customError(string.format("Block '%s' not found.", name or self.currentName))
+		end
+
+		self.currentName = "main"
+		return self:uptime(block.name)
+	end,
+
```

Comment:
No need of an extra line after `end,`. See the spec in the wiki. Remove the others as well.